docs: add Observability terms to glossary (#1349)
elastic/stack-docs: docs/en/glossary/glossary.asciidoc

include::{es-repo-dir}/glossary.asciidoc[tag=api-key-def]
--

[[glossary-apm-agent]] APM agent::
An open-source library, written in the same language as your service,
which <<glossary-instrumentation,instruments>> your code and collects performance data
and errors at runtime.
//Source: Observability

[[glossary-apm-server]] APM Server::

> An open-source application that receives data from <<glossary-apm-agent,APM agents>> and transforms

Reviewer: Not sure we want to use the word "transforms" here since we have the concept of a [transform](https://www.elastic.co/guide/en/elasticsearch/reference/7.9/transform-overview.html) in Elasticsearch. Can we say that it "sends the data to {es}" or "stores the data in {es}" instead?

> An open-source application that receives data from <<glossary-apm-agent,APM agents>> and transforms
> it into {es} documents.

Author: ```suggestion
An open-source application that receives data from <<glossary-apm-agent,APM agents>> and sends
it to {es}.
```

{ml-docs}/ml-influencers.html[Influencers].
//Source: X-Pack

[[glossary-ingestion]] ingestion::
The process of collecting and sending data from various data sources to {es}.
//Source: Observability

[[glossary-input-plugin]] input plugin::
A {ls} <<glossary-plugin,plugin>> that reads <<glossary-event,event>> data
from a specific source. Input plugins are the first stage in the {ls}
event processing <<glossary-pipeline,pipeline>>. Popular input plugins include
file, syslog, redis, and beats.
//Source: Logstash

[[glossary-instrumentation]] instrumentation::
Extending application code to track where your application is spending time.
Code is considered instrumented when it collects and reports this performance data to APM.
//Source: Observability

[[glossary-integration]] integration::

> Out-of-the-box configurations for common data sources to simplify the collection,

Reviewer: I'm not sure we can solve this problem in the glossary, but users need to understand that the integrations available for Fleet (fka Integration Manager) are different from the integrations provided by modules. WDTY?
